plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererUserModificationTracker.kt

```diff
@@ -57,6 +57,22 @@ data class AcceptedSuggestionEntry(
     val connection: ToolkitConnection?
 ) : UserModificationTrackingEntry
 
+data class CodeInsertionDiff(
+    val original: String,
+    val modified: String,
+    val diff: Double
+)
+
+fun CodeInsertionDiff?.percentage(): Double = when {
+    this == null -> 1.0
+
+    // TODO: should revisit this case
+    original.isEmpty() || modified.isEmpty() -> 1.0
```

Comment:
we may want to revisit the definition here but since the purpose of this PR is to add new metadata, will leave it as it is https://github.com/aws/aws-toolkit-jetbrains/pull/4770/files#diff-96a4362b938b115a00610e9f2095d2d9b76845fba7755a4c3a77142a54e05612L173